### ISSUE TITLE
Feed cards contain comment links with no text

### ISSUE THREAD

**Issue by aitchiss** [forem/forem#14881]
**Describe the bug**

The home feed card includes an icon for comments, and if there are currently any comments, a number indicating how many.

On mobile the accessible name for this link is either:

- Completely blank (if no comments exist)
- Just the number (e.g. a link with name '1')

On desktop the accessible names are:

- Add comment (if no comments exist)
- "1 comment" / "X comments" (if comments do exist)

Links should have meaningful text that describes exactly where the link goes - for the comment link none of the names above fits the purpose (even "Add comment" is not descriptive enough, as dozens of links with the same name exist on the page, all pointing to different pages).

We can use `aria-label` to add an accessible name to these buttons, consistent across mobile and desktop - e.g. "Comments for {article name} ({number of comments})" (happy to take suggestions on better phrasing for this!).

**To Reproduce**

- Switch on a screen reader and visit the home feed
- Try to read any of the comment links on mobile, notice there is no name for the links
- Repeat on desktop and notice many "add comment' links are announced
- Alternatively, use the browser dev tools to check the accessible name for these links 

**Expected behavior**

- All links should have an accessible name that accurately describes where it goes (i.e. comments page for the given article)

**Screenshots**

![screenshot of dev tools popup on a comment link. The link accessible name is "1"](https://user-images.githubusercontent.com/20773163/135638962-93b74e11-ecc9-48d0-8457-1c5d8e162541.png)

**Comment by pramodrhegde** [comment]
@aitchiss I would love to make a contribution here. please assign this to me. thanks!

**Comment by cmgorton** [comment]
Hey @pramodrhegde would you like to work on this issue and #14882 ? Or just one of them?

**Comment by pramodrhegde** [comment]
@cmgorton  As you suggested let me start with this one for now.

**Comment by cmgorton** [comment]
Great @pramodrhegde . I will assign this issue to you. Please let us know if you have any questions as you start to work on the issue.

**Comment by pramodrhegde** [comment]
Yeah sure, thanks @cmgorton !

**Comment by pramodrhegde** [comment]
@cmgorton I was trying to set up local and tried to log in as default admin user which is mentioned here: https://developers.forem.com/backend/database/
I am getting this error screen that its invalid creds.

<img width="1440" alt="Screenshot 2021-10-03 at 3 00 14 PM" src="https://user-images.githubusercontent.com/9020200/135747978-79dcd17d-7a5a-417b-844d-466cdef1d566.png">

need some help here on how can I log in and see some feeds data to further investigate this issue. thanks!

**Comment by rhymes** [comment]
@pramodrhegde if you start the app and have no content, it's usually because `bin/setup` hasn't worked correctly. You can't log in because the admin user is a regular user stored in the DB and created by the seed file, so if that didn't work, the user is not there, credentials included.

If you don't want to nuke your own setup, I'd suggest turning off the app and issuing `bundle exec rails db:reset` which is going to empty your DB and populate it again with the seed/mock data (you will lose your Twitter user but you can just recreate it the first time you log in). Make sure the procedure works.

Then you can start the app and you should have the initial data

**Comment by pramodrhegde** [comment]
That's definitely what might have happened there. Thanks for the help @rhymes, I'll give it a shot!

**Comment by pramodrhegde** [comment]
@rhymes Tried running the above command, but there was an error:

```
1. Creating Organizations.
rails aborted!
ActiveRecord::RecordInvalid: Validation failed: Profile image can't be blank, Profile image could not download file: Failed to open TCP connection to 2606:50c0:8003::153:443 (No route to host - connect(2) for "2606:50c0:8003::153" port 443)
```
So facing the same issue again.

**Comment by rhymes** [comment]
Seems like a networkin issue, given: 

> Failed to open TCP connection to 2606:50c0:8003::153:443 (No route to host - connect(2) for "2606:50c0:8003::153" port 443

are you behind some kind of proxy that's blocking connections? The setup process needs access to the Internet I guess
